File: app/src/main/java/com/beone/bestpractice/local/CountryDao.java
package com.beone.bestpractice.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.beone.bestpractice.local.model.CountryEntity;

import java.util.List;

@Dao
public interface CountryDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  public void insertCountrys(CountryEntity... countryEntities);

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  public void insertCountry(CountryEntity countryEntity);

  @Query("SELECT * FROM countryentity ORDER BY cases DESC")
  public LiveData<List<CountryEntity>> getCountrys();

}

File: app/src/main/java/com/beone/bestpractice/core/data/source/local/room/CountryDao.kt
package com.beone.bestpractice.core.data.local.room

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.beone.bestpractice.core.data.local.entities.CountryEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface CountryDao {
    @Query("Select * From country")
    fun getAllCountry(): Flow<List<CountryEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertCountry(countrys: List<CountryEntity>)
}